src/arrayMethodJoin.js

```diff
@@ -4,9 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += separator;
+      } else {
+        result += this[i];
+        result += separator;
+      }
+    }
```

Comment:
Better add `this[i]` to sting in case when  `(this[i] !== null || this[i] !== undefined)`.

And add one more `if` statement  for adding separator. Think what condition would you have.

This way code would be shorter and more understandable.